src/Parser/FireMage/Modules/Items/Tier20_4set.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatNumber } from 'common/format';
+
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Module from 'Parser/Core/Module';
+import getDamageBonus from '../MageCore/GetDamageBonus';
+
+const CRITICAL_MASSIVE_DAMAGE_BONUS = 0.1;
+
+class Tier20_4set extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  }
+
+  damage = 0;
+
+  on_initialized() {
+	this.active = this.combatants.selected.hasBuff(SPELLS.FIRE_MAGE_T20_4SET_BONUS_BUFF.id);
+  }
+
+  on_byPlayer_damage(event) {
+    if (this.combatants.selected.hasBuff(SPELLS.CRITICAL_MASSIVE.id)) {
+    }
+    if (event.ability.guid === SPELLS.PYROBLAST.id || event.ability.guid === SPELLS.FLAMESTRIKE.id) {
+      if (this.combatants.selected.hasBuff(SPELLS.CRITICAL_MASSIVE.id)) {
+        this.damage += getDamageBonus(event, CRITICAL_MASSIVE_DAMAGE_BONUS);
+      }
```

Comment:
Check events to make sure the damage event is consistently showing before the buff fades, sometimes WCL is weird about that.

Reply:
How would I go about that? I'm not 100% sure on that.

Reply:
easiest would be to use the debug tools wowanalyzer has, like the event list. When running the analyzer locally using 'npm start', extra tabs will be available. Click the 'Events' tab, and use the filters to show what you're looking for.

Check that the damage event is happening before the buff fades.

Reply:
the event section towards the bottom would show it. Thats a good place for figuring out the order that stuff happens